api/envoy/api/v2/listener/listener.proto

```diff
@@ -204,7 +204,18 @@ message FilterChain {
   string name = 7;
 }
 
+// Specifies the match criteria. All of the criteria needs to be satisfied.
+message ListenerFilterMatch {
+  // If it is not empty, the criteria is satisfied if the destination port of the connection is
+  // contained.
+  repeated uint32 destination_ports = 1
```

Comment:
Can you speak to the set of criteria you think eventually lives here? How similar is this to the structures we already see in `FilterChainMatch`?

Reply:
Theoretically the field set is not beyond the `FilterChainMatch`
However, the semantic could vary.  
For example, the `destination_port` in FilterChainMatch is matching SINGLE port as below. But here we need a collection of ports.

`google.protobuf.UInt32Value destination_port = 8 [(validate.rules).uint32 = {lte: 65535 gte: 1}];`

Reply:
How is everything that gets added to this structure going to work? Is this all a logical AND?

Per other comments IMO we should just go straight to a structure in which the oneof lives inside this message, like all the other matchers we have.

So this would look like:
```
message ListenerFilterMatch {
  oneof match_type {
    destination_port = 1;
    // future examples follow
    repeated ListenerFilterMatch and_matcher = 2; // for future, needs a wrapper message for oneof
    repeated ListenerFilterMatch or_matcher = 3; // for future, needs a wrapper message for oneof
  }
}
```

Reply:
I get the idea of and/or/not engine and updated 

I got trouble starting with due to `oneof` and `repeated` is exclusive in protobuf :(